### PR TITLE
Directly convert Matrix and room Ids to pills

### PR DESCRIPTION
**Pull request by weeman1337**
Should be reviewed commit-wise.

| Before | After |
| --- | --- |
| pill_user_before | pill_user |

closes https://github.com/vector-im/element-web/issues/21867

PSF-1927

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Directly convert Matrix and room Ids to pills ([\#10267](https://github.com/matrix-org/matrix-react-sdk/pull/10267)). Fixes vector-im/element-web#21867.

### CODE REVIEW

File: src/components/views/messages/TextualBody.tsx [hunk 1]
```diff
@@ -89,11 +89,8 @@ export default class TextualBody extends React.Component<IBodyProps, IState> {
         const showLineNumbers = SettingsStore.getValue("showCodeLineNumbers");
         this.activateSpoilers([this.contentRef.current]);
 
-        // pillifyLinks BEFORE linkifyElement because plain room/user URLs in the composer
```
Comment (weeman1337) [inline src/components/views/messages/TextualBody.tsx:92]: From https://github.com/vector-im/element-meta/issues/1030

> The RTE team have their own issue tracking this version of the work - we will not implement anything in the RTE, instead they will aim to match the behaviour we're implementing. As both composers will likely live alongside each other - this feels like the best approach

**Review by andybalaam — APPROVED** [review]
Looks good, but I don't understand how this relates to the comment that you are deleting.

The comment says we should do this if the composer pillifies things. The compose doesn't pillify things as you type, but are you saying we should do this anyway? (I think I agree, just want to be clear)

**Comment by weeman1337** [comment]
> The comment says we should do this if the composer pillifies things. The compose doesn't pillify things as you type, but are you saying we should do this anyway? (I think I agree, just want to be clear)

Yes. I would refer to my [comment](https://github.com/matrix-org/matrix-react-sdk/pull/10267#discussion_r1121905391) . We do now pillify it if possible. The new composer will catch up on that.

**Comment by andybalaam** [comment]
> > The comment says we should do this if the composer pillifies things. The compose doesn't pillify things as you type, but are you saying we should do this anyway? (I think I agree, just want to be clear)
> 
> Yes. I would refer to my [comment](https://github.com/matrix-org/matrix-react-sdk/pull/10267#discussion_r1121905391) in this PR. We do now pillify it if possible. The new composer will catch up on that.

Ah OK, now I understand the connection. Thanks, looks good!

**Pull request merged**